### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by andrew-kuzmin**

### CODE REVIEW

File: src/main/java/com/phoenixoft/teambalanceapp/auth/TBUserDetailsService.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@AllArgsConstructor
+public class TBUserDetailsService implements UserDetailsService {
+
+    private UserCredentialsRepository repository;
+
+    @Override
+    public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundException {
+        UserCredentials credentials = repository.findByUsername(userName)
+                .orElseThrow(() -> new UsernameNotFoundException("Username not found"));
+        return new User(credentials.getUsername(), credentials.getPassword(), new ArrayList<>());
```
Comment (JackKarichkovskiy) [inline src/main/java/com/phoenixoft/teambalanceapp/auth/TBUserDetailsService.java]: Please use Lists.EMPTY_LIST instead new ArrayList<>()

Comment (andrew-kuzmin) [reply src/main/java/com/phoenixoft/teambalanceapp/auth/TBUserDetailsService.java]: It's not applicable here.

File: src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationResponse.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.phoenixoft.teambalanceapp.auth;
+
+import java.io.Serializable;
+
+public class AuthenticationResponse implements Serializable {
```
[the remark this answers is not in the document]
Comment (andrew-kuzmin) [reply src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationResponse.java]: No need to do this. The constructor is used to create an instance.

File: src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationRequest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package com.phoenixoft.teambalanceapp.auth;
+
+import java.io.Serializable;
+
+public class AuthenticationRequest implements Serializable {
```
[the remark this answers is not in the document]
Comment (andrew-kuzmin) [reply src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationRequest.java]: Done

File: src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationRequest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    //need default constructor for JSON Parsing
+    public AuthenticationRequest()
+    {
+
+    }
```
[the remark this answers is not in the document]
Comment (andrew-kuzmin) [reply src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationRequest.java]: Done

File: src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationRequest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    public AuthenticationRequest(String username, String password) {
+        this.setUsername(username);
+        this.setPassword(password);
+    }
```
[the remark this answers is not in the document]
Comment (andrew-kuzmin) [reply src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationRequest.java]: Done

File: src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationController.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            );
+        }
+        catch (BadCredentialsException e) {
+            throw new Exception("Incorrect username or password", e);
+        }
+
+        final UserDetails userDetails = userDetailsService
+                .loadUserByUsername(authenticationRequest.getUsername());
+
+        final String jwt = jwtTokenUtil.generateToken(userDetails);
```
[the remark this answers is not in the document]
Comment (andrew-kuzmin) [reply src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationController.java]: Why do we need an additional service layer?

[the remark this answers is not in the document]
Comment (JackKarichkovskiy) [reply src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationController.java]: To extract any logic from controllers. Controllers are only web endpoints.

File: src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationResponse.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (JackKarichkovskiy) [reply src/main/java/com/phoenixoft/teambalanceapp/auth/AuthenticationResponse.java]: Value annotation will generate you constructor and getters that make your object immutable. Or what do you mean?

File: src/main/java/com/phoenixoft/teambalanceapp/auth/TBUserDetailsService.java (hunk 1 above)
Comment (JackKarichkovskiy) [reply src/main/java/com/phoenixoft/teambalanceapp/auth/TBUserDetailsService.java]: new ArrayList<>() will allocate memory every time the method is called. According to the fact that this method should be called quite frequently, this change could improve memory usage.

**Pull request merged**